src/event/invoice_payment.rs

```diff
@@ -211,36 +222,45 @@ fn hackmarket_purchase<'a>(
                 ..Default::default()
             }).map_err(|e| format!("database err: {}", e)),
             banker::pay(seller.clone(), price, paid_for),
-            market::log_blocks(vec![
-                json!({
-                    "type": "section",
-                    "text": mrkdwn(format!(
-                        "The sale of a *{}* has gone through! \
-                        <@{}> made the purchase on hackmarket, earning <@{}> *{} GP*!",
-                        name, paid_invoice.invoicee, seller, price
-                    )),
-                    "accessory": {
-                        "type": "image",
-                        "image_url": format!("http://{}/gotchi/img/{}/{}.png", *URL, category, filify(&name)),
-                        "alt_text": "Hackpheus sitting on bags of money!",
-                    }
-                }),
-                comment("U NO GET 2 BYE DAT 1"),
-            ]),
-            dm_blocks(seller.clone(), vec![
-                json!({
-                    "type": "section",
-                    "text": mrkdwn(format!(
-                        "The sale of your *{}* has gone through! \
-                        <@{}> made the purchase on hackmarket, earning you *{} GP*!",
-                        name, paid_invoice.invoicee, price
-                    )),
-                    "accessory": {
-                        "type": "image",
-                        "image_url": format!("http://{}/gotchi/img/{}/{}.png", *URL, category, filify(&name)),
-                        "alt_text": "Hackpheus sitting on bags of money!",
-                    }
-                }),
+            market::log_blocks(
+                format!("{} purchased a {} on hackmarket for {} GP!", 
+                paid_invoice.invoicee,
+                name,
+                price
+            ).to_string(),
```

Comment:
I'm not sure how you tricked `cargo fmt` into accepting this formatting, but it's heinous.